### ISSUE TITLE
[2.16.0] CoreObject observer+reads regression

### ISSUE THREAD

**Issue by ilucin** [emberjs/ember.js#15836]
Pretty weird regression happened in 2.16.0. I failed tracing down the commit that caused this so maybe you guyz can help. 

Just paste this snippet in the console of your Ember app and see for yourself:
```
 const Obj = Ember.Object.extend({
    name: Ember.computed.reads('something.else'),
    nameDidChange: Ember.observer('name', function() {})
  });

  Obj.create({name: 'obj1'});
  Obj.create({name: 'obj2'});
```

This is what happens: [Devtools sshot](http://take.ms/Kqv3F)

... or to put it verbally: "If you create more than one object with overriden computed.reads property with chained dependency and an active observer that watches that property then the world will collapse".

It's an edge-case but I need exactly this kind of behaviour in my app.

The error **won't** show up if:
- we change computed.reads to non-chained property `name: Ember.computed.reads('something')`
- if we observe the property that computed.reads points to `nameDidChange: Ember.observer('something.else', function() {})`
- if you don't override computed.reads prop in creation `Obj.create({})`
- if you use Ember version < 2.16

**Comment by bekzod** [comment]
this is fixed in #15849

**Comment by ilucin** [comment]
Confirming it's fixed in 2.17
